### ISSUE TITLE
GraphQL::ParseError when loading enum definition with value = type

### ISSUE THREAD

**Issue by arinhouck** [rmosolgo/graphql-ruby#4770]
**Describe the bug**

Currently when building a schema definition from a file it throws a  `GraphQL::ParseError` due to an enum definition with a value / field = type.

**Versions**

`graphql` version: 2.2.4
`rails` (or other framework): 6.1.7
`graphql-pro`: 1.25.1

**GraphQL schema**

```ruby
    GraphQL::Schema::BuildFromDefinition.from_definition_path(
      GraphQL::Schema,
      File.expand_path('./schema.graphql', __dir__),
      default_resolve: nil
    )
```

Problematic schema (`DocumentFilterFieldTypes.type`):
```graphql
"""The supported filter fields for documents."""
enum DocumentFilterFieldTypes {
  """Filter by document type"""
  type
}
```

**Steps to reproduce**

1. Create an enum definition with a value as `type`

2. Load the definition into a Schema object

**Expected behavior**

No parse error to be present. 

**Actual behavior**

<details>
<summary>Click to view exception backtrace</summary>

```
GraphQL::ParseError:
  Expected IDENTIFIER, actual: TYPE ("type") at [2474, 3] (/Users/username/Documents/github/some_project/lib/gem_name/federation/schema.graphql)
# ./lib/gem_name/federation/schema.rb:7:in `<module:Federation>'
# ./lib/gem_name/federation/schema.rb:6:in `<module:KnowdeGraphQL>'
# ./lib/gem_name/federation/schema.rb:5:in `<top (required)>'
# ./spec/lib/gem_name/client/client_helper_spec.rb:5:in `require'
# ./spec/lib/gem_name/client/client_helper_spec.rb:5:in `<top (required)>'
```

</details>

**Additional context**

The bug was introduced when I went to upgrade from `2.0.14` to `2.2.4`.

This was not an issue in the past version. When removing the type from the enum definition the parser passed without issue.

**Comment by rmosolgo** [comment]
Hey, thanks for the detailed report! This was an oversight in hand-writing the new parser, sorry 😅  I've got a fix up at #4772

**Comment by arinhouck** [comment]
Awesome! @rmosolgo thank you for the quick response and fix! 🙏
